protocol/binary/writer.go

```diff
@@ -201,32 +168,32 @@ func (bw *Writer) WriteValue(v wire.Value) error {
 	switch v.Type() {
 	case wire.TBool:
 		if v.GetBool() {
-			return bw.writeByte(1)
+			return bw.sw.writeByte(1)
 		}
-		return bw.writeByte(0)
+		return bw.sw.writeByte(0)
```

Comment:
This should be able to leverage`bw.sw.WriteBool` now, I think.

Reply:
+1

    return bw.sw.WriteBool(v.GetBool())